docs/how-to/namespace-transition.md

```diff
@@ -16,85 +16,321 @@ To avoid these issue we recommend taking the proactive approach described in
+Pay special attention to copying `config-gcp-auth` since this contains
+information relating to authentication and may lead to disruptions in service if
+not present in the new namespace.
+
+```shell script
+# copy the existing config maps to maintain existing configurations
+kubectl get configmaps -n cloud-run-events --field-selector metadata.name!=kube-root-ca.crt -o yaml | \
+  sed 's/  namespace: cloud-run-events/  namespace: events-system/g' | \
+  sed 's!/cloud-run-events/!/events-system/!g' | \
+  kubectl apply -f -
+```
+
+The expected output is:
+
+```shell script
+configmap/config-gcp-auth created
+configmap/config-leader-election created
+configmap/config-logging created
+configmap/config-observability created
+configmap/config-tracing created
+configmap/default-brokercell-broker-targets created
+```
+
+Follow the upgrade instructions for either
+[Kubernetes Secrets](#using-kubernetes-secrets) or
+[Workload Identity](#using-workload-identity) depending on your cluster's
+current authentication mechanism. After the upgrade is completed, verify that
+everything is working correctly by following the steps in
+[Verify and Correct Existing Functionality](#verify-and-correct-existing-functionality).
+
+#### Using Kubernetes Secrets
+
+1. Setup authentication for the new namespace you created above by copying
+   existing secrets. By doing this step before the upgrade you will ensure the
+   speediest creation of the necessary resources.
 
    ```shell script
-   # create the new namespace by copying the labels and annotations from the old one
-   kubectl get namespace cloud-run-events -o yaml | \
-     sed 's/  name: cloud-run-events/  name: events-system/g' | \
-     kubectl create -f -
-   # setup identical authentication by copying the existing secrets from the old namespace
-   kubectl get secrets -n cloud-run-events -o yaml | \
-     sed 's/  namespace: cloud-run-events/  namespace: events-system/g' | \
-     kubectl create -f -
-   # copy the existing config maps to maintain existing configurations
-   kubectl get configmaps -n cloud-run-events --field-selector metadata.name!=kube-root-ca.crt -o yaml | \
-     sed 's/  namespace: cloud-run-events/  namespace: events-system/g' | \
-     kubectl create -f -
-
-   # remove the old webhook and controller to prevent them from interfering with the new ones
-   kubectl delete deployment webhook -n cloud-run-events
-   kubectl delete service webhook -n cloud-run-events
-   kubectl delete deployment controller -n cloud-run-events
-   kubectl delete service controller -n cloud-run-events
+   # auth for the control plane
+   kubectl get secret google-cloud-key --namespace=cloud-run-events -o yaml | \
+     grep -v '^\s*namespace:\s' | kubectl apply --namespace=events-system -f -
+   # auth for the broker data plane
+   kubectl get secret google-broker-key --namespace=cloud-run-events -o yaml | \
+     grep -v '^\s*namespace:\s' | kubectl apply --namespace=events-system -f -
```

Comment:
Make sure to also copy the Secret `google-cloud-sources-key`, if it is present.

That is the secret used by gcloud with `gcloud events init-namespace --copy-default-secret`.